lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,326 @@
+- name: Deploy an application in standalone mode
+  wildfly:
+    src: /tmp/hello-1.0-SNAPSHOT.war
+    deployment: hello.war
+    state: present
+    controller_address: http://localhost:9990
+    username: username
+    password: password
+    mode: standalone
+"""
+
+RETURN = '''
+---
+state:
+  description: State of the deployment.
+  returned: success
+  type: string
+  sample: deployed
+name:
+  description: Name of the deployed content.
+  returned: success
+  type: string
+  sample: hello.war
+'''
+
+HAS_JSON = True
+
+try:
+    import json
+except ImportError:
+    try:
+        import simplejson as json
+    except ImportError:
+        HAS_JSON = False
+
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import *
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def undeploy(params):
+    deployment = params['deployment']
+    remove_content = \
+        {'operation': 'remove',
+         'address': [{'deployment': deployment}],
+         'json.pretty': 1}
+    mode = params['mode']
+
+    if mode == 'domain':
+        deployed_domain, info = is_deployed_domain(params)
+    	if info['status'] == 200:
+            server_groups = params['server_groups']
+            remove_from_server = \
+                {'operation': 'remove',
+                 'address': [{'server-group': server_groups},
+                             {'deployment': deployment}],
+                 'json.pretty': 1}
+            removed, info = get_response(remove_from_server)
+            if info['status'] != 200:
+            	module.fail_json(msg='Undeploy failed', details=info['msg'])
+    return get_response(remove_content)
+
+def deploy(params):
+    deployment = params['deployment']
+    server_groups = params['server_groups']
+    add_content = \
+        {'operation': 'add',
+         'address': [{'deployment': deployment}],
+         'content': [{'url': 'file:' + params['src'] + ''}],
+         'json.pretty': 1}
+    mode = params['mode']
+
+    response_add_content, info = get_response(add_content)
+
+    if info['status'] == 200:
+        if mode == 'domain':
+            add_content_server_group = \
+                {'operation': 'add',
+                 'address': [{'server-group': server_groups},
+                             {'deployment': deployment}],
+                 'content': [{'url': 'file:' + params['src'] + ''}],
+                 'json.pretty': 1}
+            deploy_content_domain = \
+                {'operation': 'deploy',
+                 'address': [{'server-group': server_groups},
+                             {'deployment': deployment}],
+                 'json.pretty': 1}
+            response_add_content_server_group, info = \
+                get_response(add_content_server_group)
+            if info['status'] == 200:
+                return get_response(deploy_content_domain)
+            else:
```

Comment:
This is never evaluated because you are already in the branch which matches `info['status'] == 200` only.